test/cctest/node_test_fixture.h

```diff
@@ -28,13 +29,35 @@ class ArrayBufferAllocator : public v8::ArrayBuffer::Allocator {
   }
 };
 
-class NodeTestFixture : public ::testing::Test {
+struct Argv {
  public:
-  static char** createargv(const char* prog,
-                           const char* arg1,
-                           const char* arg2);
-  static void freeargv(char** argv);
+  Argv(const char* prog, const char* arg1, const char* arg2) {
```

Comment:
@bnoordhuis This not what you suggested but I'm going to try it the way you suggested. Previously I had issue on other platforms (than mac) where memory allocation for argv was not continuous leading to an error in libuv. But like I said I'll take a stab at this again.